File: front/src/types/generation/task.ts
import type { Content } from './content';

export const TaskStatus = {
    Pending: "Pending",
    Success: "Success",
    Failed: "Failed"
} as const;

export type TaskStatus = (typeof TaskStatus)[keyof typeof TaskStatus];

export interface Task {
    id: string;
    userId: string;
    prompt?: string | null;
    creationDate: string;
    contentId: string | null;
    status: TaskStatus;
    content?: Content | null;
}

File: backend/VideoBackend.Domain/Entities/Task/Task.cs
using VideoBackend.Domain.Enums;

namespace VideoBackend.Domain.Entities.Task;

public class Task
{
    public Guid Id { get; set; }
    public string? Prompt { get; set; }
    public DateTime? CreationDate { get; set; }
    public Guid UserId { get; set; }
    public User.User? User { get; set; }
    public Guid ContentId { get; set; }
    public Content? Content { get; set; }
    public GenerationStatus Status { get; set; }
    public string? PoyoTaskId { get; set; }
    public int CreditsSpent { get; set; }
}
